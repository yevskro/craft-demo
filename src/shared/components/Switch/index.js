import React from 'react';
import styled from 'styled-components';

function Switch({ left, right, onLeftSelected, onRightSelected }) {
  function handleSwitch(e) {
    const { checked } = e.target;
    if (checked) onRightSelected();
    else onLeftSelected();
  }

  return (
    <StyledSwitch>
      <HiddenCheckbox onClick={(e) => handleSwitch(e)} />
      <Text>{left}</Text>
      <Text>{right}</Text>
    </StyledSwitch>
  );
}

const StyledSwitch = styled.label`
  > input {
    display: none;
  }
`;

const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
  &:not(:checked) + span {
    background-color: #2196f3;
  }

  &:checked + span + span {
    background-color: #2196f3;
  }
`;

const Text = styled.span`
  padding: 0px 5px 0px 5px;
`;
export default Switch;
