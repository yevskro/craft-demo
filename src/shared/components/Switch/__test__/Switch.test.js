import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Switch from '../index';

test('renders without crashing', () => {
  render(<Switch />);
});

test('renders left and right tabs', () => {
  render(<Switch left="left" right="right" />);
  expect(screen.getByText('left')).toBeInTheDocument();
  expect(screen.getByText('right')).toBeInTheDocument();
});

test('invokes onLeftSelected and onRightSelected', () => {
  const onLeftMock = jest.fn();
  const onRightMock = jest.fn();
  render(
    <Switch
      left="left"
      right="right"
      onLeftSelected={onLeftMock}
      onRightSelected={onRightMock}
    />
  );

  /* 
    doesn't matter what text we select, odd amount of clicks should call
    onRightSelected, and even amount of clicks
    should call onLeftSelected, good to check that both tabs handle clicks
  */
  userEvent.click(screen.getByText('right'));
  expect(onRightMock).toHaveBeenCalledTimes(1);
  userEvent.click(screen.getByText('left'));
  expect(onLeftMock).toHaveBeenCalledTimes(1);
});
