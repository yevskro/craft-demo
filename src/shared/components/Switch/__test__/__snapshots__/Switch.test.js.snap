// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  box-shadow: 2px 5px;
}

.c0 > input {
  display: none;
}

.c0 input + span {
  border: 2px solid black;
}

.c0 span + span {
  border-top: 2px solid black;
  border-right: 2px solid black;
  border-bottom: 2px solid black;
}

.c1:not(:checked) + span {
  background-color: #2196f3;
}

.c1:checked + span + span {
  background-color: #2196f3;
}

.c2 {
  padding: 0px 5px 0px 5px;
}

<label
  className="c0"
>
  <input
    className="c1"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="c2"
  >
    left
  </span>
  <span
    className="c2"
  >
    right
  </span>
</label>
`;
